### PR TITLE
Add Service Usage API

### PR DESCRIPTION
**Pull request by hshin-g**
Addresses Forseti Security Github Issue [#3214](https://github.com/forseti-security/forseti-security/issues/3214).

Linked to Forseti Security Github PR [#3329](https://github.com/forseti-security/forseti-security/pull/3229).

--

Add Service Usage API.
Update after submodules rollback.
Supersedes #269.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -56,6 +56,7 @@ locals {
     "cloudtrace.googleapis.com",
     "container.googleapis.com",
     "servicemanagement.googleapis.com",
```
Comment (kevensen) [inline main.tf]: Do we remove this API from the list?

Comment (kevensen) [reply main.tf]: Or would doing so be a breaking change?

File: main.tf [hunk 2]
```diff
@@ -158,6 +159,9 @@ module "server" {
   servicemanagement_period                            = var.servicemanagement_period
   servicemanagement_max_calls                         = var.servicemanagement_max_calls
   servicemanagement_disable_polling                   = var.servicemanagement_disable_polling
```
Comment (kevensen) [inline main.tf]: Depending on the answer above, same here?

File: modules/rules/templates/rules/enabled_apis_rules.yaml [hunk 3]
```diff
@@ -36,6 +36,7 @@
 #       - 'replicapoolupdater.googleapis.com'
 #       - 'resourceviews.googleapis.com'
 #       - 'servicemanagement.googleapis.com'
```
Comment (kevensen) [inline modules/rules/templates/rules/enabled_apis_rules.yaml]: Remove?

File: modules/server/main.tf [hunk 4]
```diff
@@ -177,6 +177,9 @@ data "template_file" "forseti_server_config" {
     SERVICEMANAGEMENT_PERIOD                            = var.servicemanagement_period
     SERVICEMANAGEMENT_MAX_CALLS                         = var.servicemanagement_max_calls
     SERVICEMANAGEMENT_DISABLE_POLLING                   = var.servicemanagement_disable_polling
```
Comment (kevensen) [inline modules/server/main.tf]: Remove?

File: main.tf (hunk 1 above)
Comment (hshin-g) [reply main.tf]: We will leave this API in just in case we need to go back to it again.

**Comment by gkowalski-google** [comment]
@hshin-g Is there anything left to do with this PR?

**Comment by hshin-g** [comment]
@gkowalski-google We have to wait until this is merged into the Forseti side in the 2.23.0 release.

**Pull request merged**